### PR TITLE
add packaging to required packages

### PR DESCRIPTION
**Pull request by Gabriel-p**
### Description

Package `packaging` is missing from the required list

EDIT: This is a direct follow-up of #11714 .

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by pllim** [comment]
I know I said no change log earlier but now I am not sure... @saimn ?

**Comment by saimn** [comment]
Probably better to have one, for the people packaging astropy.

**Comment by pllim** [comment]
@Gabriel-p , can you please add a change log in `astropy/docs/changes` named `12199.other.rst`?

Also see: https://github.com/astropy/astropy/blob/main/docs/changes/README.rst

**Comment by Gabriel-p** [comment]
@pllim If I just create the file and push the commit, will it appear in this PR automatically?

File: docs/changes/12199.other.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Add missing package ``packaging``  to ``setup.cfg``.
```
Comment (pllim) [inline docs/changes/12199.other.rst:1]: It is not missing from the last release, but rather an unreleased patch now requires it, so we need to reword. 😸 

```suggestion
``astropy`` now requires ``packaging``.
```

**Comment by pllim** [comment]
Ops, also need to add it under "Requirements" at https://github.com/astropy/astropy/blob/main/docs/install.rst 😅

**Comment by Gabriel-p** [comment]
> Ops, also need to add it under "Requirements" at https://github.com/astropy/astropy/blob/main/docs/install.rst

Mmmm I'm going to need more directions to properly include the package in this file. Something like this?

```
Requirements
============

``astropy`` has the following strict requirements:

- `Python`_ |minimum_python_version| or later

- `Numpy`_ |minimum_numpy_version| or later

- `PyERFA`_ |minimum_pyerfa_version| or later

- `PyYAML <https://pyyaml.org>`_ |minimum_pyyaml_version| or later

- `packaging <https://packaging.pypa.io/>`_ |minimum_packaging_version| or later
```

> Also can remove duplicate mentions of `packaging` in `setup.cfg` (xref #11499)

Should I remove all 4 entries in this file?

**Comment by pllim** [comment]
> Something like this?

Yes, I think so.

> all 4 entries

I only see 3 entries in `main`, but yes, they are not necessary once you add it to `install_requires`.

**Comment by Gabriel-p** [comment]
I'm having trouble squashing. Here's the output of `git log`

Captura de pantalla de 2021-09-27 17-31-18

and after rebasing:

```
$ git rebase --interactive HEAD~5
Successfully rebased and updated refs/heads/add_package.
```

I get the exact same log. What did I mess up?

**Comment by pllim** [comment]
An editor should have popped up and you would pick the first one (`pick`) and squash the other ones (`s`). If you don't do anything in that step, the default is to keep all the commits unchanged. Hope this helps.

**Comment by pllim** [comment]
Hmm the merge commit should not be there. Do you want me to squash it for you? 😅

**Comment by Gabriel-p** [comment]
Go ahead please

File: docs/install.rst [hunk 2]
```diff
@@ -141,6 +141,8 @@ Requirements
 
 - `PyYAML <https://pyyaml.org>`_ |minimum_pyyaml_version| or later
 
+- `packaging <https://packaging.pypa.io/>`_ |minimum_packaging_version| or later
```
Comment (eerovaher) [inline docs/install.rst:144]: This is breaking the documentation build because `|minimum_packaging_version|` isn't defined anywhere.

**Comment by Gabriel-p** [comment]
> This is breaking the documentation build because |minimum_packaging_version| isn't defined anywhere.

Should I remove `|minimum_packaging_version|` or define it somewhere?



> At the moment this pull request is not adding `packaging` as an explicit requirement to `astropy`, it is instead removing `packaging` from `setup.cfg` entirely.

This was my mistake. I removed *all* mentions of `packaging` instead of just the redundant ones. I'll add it back now.

**Comment by pllim** [comment]
I rebased and squashed for you. I think this looks good now, just waiting for CI.

> Should I remove `|minimum_packaging_version|` or define it somewhere?

Don't worry about that directive. It is handled automatically within `docs/conf.py`.

**Comment by saimn** [comment]
21.0 is the latest version from July, I don't think we need to enforce such a recent version. Maybe 20.0 (2020-01-06), or even 19.0  (2019-01-20) ?

> Don't worry about that directive. It is handled automatically within docs/conf.py.

It should be defined in `docs/common_links.txt`, the docs build is currently failing because of that.

**Comment by Gabriel-p** [comment]
> It should be defined in `docs/common_links.txt`

Something like this perhaps?

```
.. pyyaml
.. |minimum_pyyaml_version| replace:: {pyyaml}

.. packaging
.. packaging: https://packaging.pypa.io/
.. |minimum_packaging_version| replace:: {packaging}

.. IPython
.. _IPython: http://ipython.org/
.. _`ipython github`: https://github.com/ipython/ipython
.. _`ipython mailing list`: http://mail.python.org/mailman/listinfo/IPython-dev
.. |minimum_ipython_version| replace:: {ipython}
```

**Comment by eerovaher** [comment]
You should try building the documentation locally to see if everything works. You can follow these instructions: https://docs.astropy.org/en/latest/install.html#builddocs

But it looks like the line
```
.. packaging: https://packaging.pypa.io/
```
is missing an underscore.

**Comment by pllim** [comment]
> or even 19.0 (2019-01-20)

Yes, that is probably safer.

> links

Ops... I totally forgot there's one more file.

@Gabriel-p , so there are two more things. If you need help with pulling this branch back to your local machine, please let me know. It is no longer in sync with your original branch that you started with. I'll be happy to wrap this up for you too if you are getting sick of `packaging`. 😅

**Comment by pllim** [comment]
Okay, no problem. I'll copy paste your patch from the comment and give you co-author credit. Thank you for your effort! I thought this was going to be a one-liner but clearly I was wrong...

**Comment by pllim** [comment]
Since this adds a new required dependency, I also enabled cron jobs, just in case. I'll cancel pyinstaller one because it is broken for unrelated reason already.

**Comment by pllim** [comment]
Thanks, all!

**Pull request merged**